### PR TITLE
refactor: modify migrated integration test format to be a guide for the upcoming works

### PR DESCRIPTION
**Pull request by devjiwonchoi**
As stated in https://github.com/huozhi/bunchee/pull/455#discussion_r1493786108 and https://github.com/huozhi/bunchee/pull/455#discussion_r1493786266, this PR modifies the migrated integration tests for later works.

Extends PR #455
Extends Issue #449

### CODE REVIEW

File: test/integration/default-node-mjs/index.test.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { readFile } from 'fs/promises'
 import { createIntegrationTest, existsFile } from '../utils'
 
-describe('integration', () => {
-  test(`default-node-mjs`, async () => {
+describe('integration default-node-mjs', () => {
+  it('should work on default .node.mjs convention', async () => {
```
Comment (huozhi) [inline test/integration/default-node-mjs/index.test.ts:5]:
```suggestion
  it('should work with .mjs extension', async () => {
```

File: test/integration/dev-prod-convention/index.test.ts [hunk 2]
```diff
@@ -1,11 +1,7 @@
...
+import { createIntegrationTest, assertFilesContent } from '../utils'
 
-afterEach(async () => {
-  await deleteFile(`${__dirname}/fixtures/tsconfig.json`)
-})
-
-describe('integration', () => {
-  test(`dev-prod-convention`, async () => {
+describe('integration dev-prod-convention', () => {
+  it('should work on environment conventions', async () => {
```
Comment (huozhi) [inline test/integration/dev-prod-convention/index.test.ts:4]:
```suggestion
  it('should work on optimize conventions dev and prod', async () => {
```

File: test/integration/edge-variable/index.test.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { createIntegrationTest, assertFilesContent } from '../utils'
 
-describe('integration', () => {
-  test(`edge-variable`, async () => {
+describe('integration edge-variable', () => {
+  it('should work on .edge convention', async () => {
```
Comment (huozhi) [inline test/integration/edge-variable/index.test.ts:4]:
```suggestion
  it('should work with edge export condition', async () => {
```

File: test/integration/entry-index-index/index.test.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { createIntegrationTest, assertFilesContent } from '../utils'
 
-describe('integration', () => {
-  test(`entry-index-index`, async () => {
+describe('integration entry-index-index', () => {
+  it('should work on index file inside index directory', async () => {
```
Comment (huozhi) [inline test/integration/entry-index-index/index.test.ts:4]:
```suggestion
  it('should work with index file inside index directory', async () => {
```

File: test/integration/esm-pkg-cjs-main-field/index.test.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { createIntegrationTest, assertContainFiles } from '../utils'
 
-describe('integration', () => {
-  test(`esm-pkg-cjs-main-field`, async () => {
+describe('integration esm-pkg-cjs-main-field', () => {
+  it('should work on ESM package, main field as CJS ', async () => {
```
Comment (huozhi) [inline test/integration/esm-pkg-cjs-main-field/index.test.ts:4]:
```suggestion
  it('should work with ESM package with CJS main field ', async () => {
```

File: test/integration/esm-shims/index.test.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { createIntegrationTest, assertFilesContent } from '../utils'
 
-describe('integration', () => {
-  test(`esm-shims`, async () => {
+describe('integration esm-shims', () => {
+  it('should work on ESM shims', async () => {
```
Comment (huozhi) [inline test/integration/esm-shims/index.test.ts:4]:
```suggestion
  it('should work with ESM shims', async () => {
```

File: test/integration/externals/index.test.ts [hunk 7]
```diff
@@ -1,8 +1,8 @@
 import { readFile } from 'fs/promises'
 import { createIntegrationTest } from '../utils'
 
-describe('integration', () => {
-  test(`externals`, async () => {
+describe('integration externals', () => {
+  it('should work on externals', async () => {
```
Comment (huozhi) [inline test/integration/externals/index.test.ts:5]:
```suggestion
  it('should handle externals', async () => {
```

File: test/integration/js-only/index.test.ts [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import { createIntegrationTest, existsFile } from '../utils'
 
-describe('integration', () => {
-  test(`js-only`, async () => {
+describe('integration js-only', () => {
+  it('should work on JS only', async () => {
```
Comment (huozhi) [inline test/integration/js-only/index.test.ts:4]:
```suggestion
  it('should work with js only project', async () => {
```

**Review by huozhi — APPROVED** [review]
Thanks!

**Comment by devjiwonchoi** [comment]
It's my pleasure. Thank you!

**Pull request merged**